site/book/04-using-functions/01-declarative-function-execution.md

```diff
@@ -178,5 +178,43 @@ pipeline:
         tier: mysql
 ```
 
+### `selectors`
+
+Selectors can be used to target specific resources for function execution. Selectors 
+follow OR of AND(s) approach where, within each selector, the filters are ANDed 
+and the selected resources are UNIONed with other selected resources. Please go through 
+[resource selector guide] for more information.
+
+Here are the list of available selector properties:
+
+1. `apiVersion`: `apiVersion` field value of resources to be selected.
+2. `kind`: `kind` field value of resources to be selected.
+3. `name`: `metadata.name` field value of resources to be selected.
+4. `namespace`: `metadata.namespace` field of resources to be selected.
+5. `packagePath`: PackagePath of resources to be selected. The path must be 
+   OS-agnostic Slash-separated relative to the package directory. Examples
+   - `packagePath: ./mysql` - selects resources in the `mysql` subpackage excluding resources of nested subpackages of `mysql`.
+   - `packagePath: ./` - selects resources in current package excluding resources in subpackages of current package.
+   
+For example:
```

Comment:
How the function works shouldn't be left as an exercise to the reader. This is THE most important part.